packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -0,0 +1,1486 @@
+      //       <ComboBox label="Test" onOpenChange={onOpenChange} onFilter={onFilter} items={[]}>
+      //         {(item) => <Item key={item.key}>{item.name}</Item>}
+      //       </ComboBox>
+      //     </Provider>
+      //   );
+
+      //   let button = getByRole('button');
+      //   act(() => {
+      //     triggerPress(button);
+      //     jest.runAllTimers();
+      //   });
+
+      //   expect(onFilter).toHaveBeenCalled();
+
+      //   expect(onOpenChange).toHaveBeenCalledWith(true);
+      //   expect(onOpenChange).toHaveBeenCalledTimes(1);
+
+      //   expect(() => getByRole('listbox')).toThrow();
+      // });
+
+      it('focuses first item if there are items loaded', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        act(() => {
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        expect(onOpenChange).toHaveBeenCalledWith(true);
+
+        let listbox = getByRole('listbox');
+        testComboBoxOpen(combobox, button, listbox, 0);
+      });
+
+      it('opens for touch', () => {
+        let {getByRole} = renderComboBox({});
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(document.activeElement).not.toBe(combobox);
+
+        act(() => {
+          fireEvent.touchStart(button, {targetTouches: [{identifier: 1}]});
+          fireEvent.touchEnd(button, {changedTouches: [{identifier: 1, clientX: 0, clientY: 0}]});
+          jest.runAllTimers();
+        });
+
+        expect(document.activeElement).toBe(combobox);
+        let listbox = getByRole('listbox');
+        expect(listbox).toBeTruthy;
+        expect(document.activeElement).toBe(combobox);
+      });
+    });
+
+    describe('keyboard input', function () {
+      it('opens the menu on down arrow press', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        combobox.focus();
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          fireEvent.keyDown(combobox, {key: 'ArrowDown', code: 40, charCode: 40});
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        testComboBoxOpen(combobox, button, listbox, 0);
+      });
+
+      it('opens the menu on up arrow press', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        combobox.focus();
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          fireEvent.keyDown(combobox, {key: 'ArrowUp', code: 38, charCode: 38});
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        testComboBoxOpen(combobox, button, listbox, 2);
+      });
```

Comment:
Do we need to test other keys for open events like pagedown, home, etc.? Should it be done with 
```
it.each`
    Name            | keyObject
    ${'ArrowUp'} | ${{key: 'ArrowDown', code: 40, charCode: 40}}
    ${'ArrowUp'} | ${{key: 'ArrowUp', code: 38, charCode: 38}}
  `

Reply:
I don't believe pagedown/home/etc are commands to open the combobox, those are listbox specific commands